experimental/integrations/prototype-spark/marquez-spark-agent/src/main/java/marquez/spark/agent/lifecycle/SparkSQLExecutionContext.java

```diff
@@ -1,184 +1,187 @@
+            .namespace(marquezContext.getJobNamespace())
+            .name(rootPath.toUri().toString().replaceAll(":", "_"));
+        inputDatasets.add(lineageDataset.build());
       }
     }
-    extraInfo.append(prefix + plan.getClass().getSimpleName() + ":" + extra + " plan: " + plan).append("\n");
-    Collection<LogicalPlan> children = JavaConversions.asJavaCollection(plan.children());
-    for (LogicalPlan child : children) {
-      print(prefix + "  ", child);
-    }
+
+    return inputDatasets;
   }
 
-  public void end(SparkListenerSQLExecutionEnd endEvent) {
-    logger.info("Spark sql execution ended " + endEvent);
-    if (success == null) {
-      extraInfo.append("unknown end status\n");
-    } else if (success) {
-      marquezContext.success(runId, jobId, endEvent.time(), outputs);
-    } else  {
-      marquezContext.failure(runId, jobId, endEvent.time(), outputs, error );
+  private List<LineageDataset> buildOutputs(LogicalPlan logical) {
+    List<LineageDataset> inputDatasets = new ArrayList<>();
+    if (logical instanceof InsertIntoHadoopFsRelationCommand) {
+      InsertIntoHadoopFsRelationCommand insert = (InsertIntoHadoopFsRelationCommand)logical;
+      inputDatasets.add(LineageDataset.builder()
+          .namespace(marquezContext.getJobNamespace())
+          .name(insert.outputPath().toUri().toString())
+          .build());
     }
-    logger.info("job ended: " + endEvent + " \n" + extraInfo.toString());
+    return inputDatasets;
   }
 
-  @Override
-  public void setActiveJob(ActiveJob activeJob) {
-    jobId = activeJob.jobId();
-    RDD<?> finalRdd = activeJob.finalStage().rdd();
-    logger.info("Registered Active job as part of spark-sql:" + activeJob.jobId() + " rdd: " + finalRdd);
-    Set<RDD<?>> rdds = Rdds.flattenRDDs(finalRdd);
-    for (RDD<?> rdd : rdds) {
-      Configuration config = SparkListener.removeConfigForRDD(rdd);
-      if (config != null) {
-        extraInfo.append("unexpected rdd => conf mapping\n  ").append(rdd).append(" => ").append(config).append("\n");
-        logger.warn("This should not be here: " + rdd + " => " + config);
-      }
-    }
+  private LineageRun buildRun(Object logicalPlanFacet, Object physicalPlanFacet) {
+    return LineageRun.builder()
+        .runId(marquezContext.getParentRunId())
+        .facets(
+            RunFacet.builder()
+                .additional(ImmutableMap.of(
+                    LogicalPlan.class.getName(), logicalPlanFacet,
+                    SparkPlan.class.getName(), physicalPlanFacet)
+                ).build())
+        .build();
   }
 
-  @Override
-  public void start(SparkListenerJobStart jobStart) {
-    logger.info("Starting job as part of spark-sql:" + jobStart.jobId());
-    extraInfo.append(Rdds.toString(jobStart));
+  private Object buildLogicalPlanFacet(LogicalPlan plan) {
+    try {
+      return marquezContext.getMapper().readTree(plan.toJSON());
```

Comment:
nice

Reply:
we may want to check that this representation doesn't leave too much information out.